### PR TITLE
Inject user controller into UserAuthenticationController.

### PR DESCRIPTION
**Pull request by nlutsenko**
Don't get global via `PFUser.+userController`, but inject it via `dataSource`.

### CODE REVIEW

File: Parse/Internal/User/AuthenticationProviders/Controller/PFUserAuthenticationController.m [hunk 1]
```diff
@@ -113,12 +113,12 @@ - (void)unregisterAuthenticationDelegateForAuthType:(NSString *)authType {
...
 
-- (BFTask *)logInUserAsyncWithAuthType:(NSString *)authType authData:(NSDictionary *)authData {
+- (BFTask PF_GENERIC(PFUser *)*)logInUserAsyncWithAuthType:(NSString *)authType
+                                                  authData:(NSDictionary PF_GENERIC(NSString *, NSString *)*)authData {
     //TODO: (nlutsenko) Make it fully async.
-    //TODO: (nlutsenko) Inject `PFUserController` here.
     PFUser *currentUser = [PFUser currentUser];
     if (currentUser && [PFAnonymousUtils isLinkedWithUser:currentUser]) {
-        if ([currentUser isLazy]) {
+        if (currentUser.isLazy) {
```
Comment (richardjrossiii) [inline Parse/Internal/User/AuthenticationProviders/Controller/PFUserAuthenticationController.m]: Shouldn't it be `.lazy`?

File: Parse/Internal/BFTask+Private.h [hunk 2]
```diff
@@ -22,19 +22,19 @@
 
 @interface BFTask (Private)
 
-- (instancetype)continueAsyncWithBlock:(BFContinuationBlock)block;
-- (instancetype)continueAsyncWithSuccessBlock:(BFContinuationBlock)block;
+- (BFTask *)continueAsyncWithBlock:(BFContinuationBlock)block;
```
Comment (richardjrossiii) [inline Parse/Internal/BFTask+Private.h]: Separate PR? I don't feel strongly however.

File: Parse/Internal/User/AuthenticationProviders/Controller/PFUserAuthenticationController.m (hunk 1 above)
Comment (nlutsenko) [inline Parse/Internal/User/AuthenticationProviders/Controller/PFUserAuthenticationController.m]: I thought so as well, but property is 'isLazy'

File: Parse/Internal/BFTask+Private.h (hunk 2 above)
Comment (nlutsenko) [inline Parse/Internal/BFTask+Private.h]: Nah, too much work.

**Pull request merged**